packages/instrumenter/src/mutators/array-declaration-mutator.ts

```diff
@@ -1,25 +1,76 @@
+        const replacement = arrayDeclarationReplacements.FilledArray.replacementOperator;
+        yield replacement;
+      } else if (operations.includes(arrayDeclarationReplacements.EmptyArray.mutatorName as string) && !path.node.elements.length) {
+        const replacement = arrayDeclarationReplacements.EmptyArray.replacementOperator;
+        yield replacement;
+      }
     }
-    if ((path.isCallExpression() || path.isNewExpression()) && types.isIdentifier(path.node.callee) && path.node.callee.name === 'Array') {
-      const mutatedCallArgs = path.node.arguments.length ? [] : [types.arrayExpression()];
-      const replacement = types.isNewExpression(path.node)
-        ? types.newExpression(deepCloneNode(path.node.callee), mutatedCallArgs)
-        : types.callExpression(deepCloneNode(path.node.callee), mutatedCallArgs);
-      yield replacement;
+    // Check for the new Array() construct in code
+    if (
+      (path.isCallExpression() || path.isNewExpression()) &&
+      types.isIdentifier(path.node.callee) &&
+      path.node.callee.name === 'Array' &&
+      (operations == undefined ||
+        operations.includes(arrayDeclarationReplacements.EmptyArrayConstructor.mutatorName as string) ||
+        operations.includes(arrayDeclarationReplacements.FilledArrayConstructor.mutatorName as string))
+    ) {
+      const emptyArrayReplacement: Array<ArgumentPlaceholder | Expression | JSXNamespacedName | SpreadElement> =
+        arrayDeclarationReplacements.EmptyArrayConstructor.replacementOperator;
+      const FilledArrayReplacement: Array<ArgumentPlaceholder | Expression | JSXNamespacedName | SpreadElement> =
+        arrayDeclarationReplacements.FilledArrayConstructor.replacementOperator;
+      const mutatedCallArgs = path.node.arguments.length ? FilledArrayReplacement : emptyArrayReplacement;
+
+      if (operations == undefined) {
+        const replacement = types.isNewExpression(path.node)
+          ? types.newExpression(deepCloneNode(path.node.callee), mutatedCallArgs)
+          : types.callExpression(deepCloneNode(path.node.callee), mutatedCallArgs);
+        yield replacement;
+      } else if (operations.includes(arrayDeclarationReplacements.FilledArrayConstructor.mutatorName as string) && path.node.arguments.length) {
+        const replacement = types.isNewExpression(path.node)
+          ? types.newExpression(deepCloneNode(path.node.callee), mutatedCallArgs)
+          : types.callExpression(deepCloneNode(path.node.callee), mutatedCallArgs);
+        yield replacement;
+      } else if (operations.includes(arrayDeclarationReplacements.EmptyArrayConstructor.mutatorName as string) && !path.node.arguments.length) {
+        const replacement = types.isNewExpression(path.node)
+          ? types.newExpression(deepCloneNode(path.node.callee), mutatedCallArgs)
+          : types.callExpression(deepCloneNode(path.node.callee), mutatedCallArgs);
+        yield replacement;
```

Comment:
The block of this if statement is the same in all three cases right? Then, maybe you could merge the if's boolean expressions, or create a separate function to improve readability